ProcessMaker/Serializers/ApiSerializer.php

```diff
@@ -1,71 +1,28 @@
 <?php
-
 namespace ProcessMaker\Serializers;
 
-use Illuminate\Http\Request;
-use League\Fractal\Pagination\CursorInterface;
-use League\Fractal\Pagination\PaginatorInterface;
-use League\Fractal\Resource\ResourceInterface;
-use League\Fractal\Serializer\SerializerAbstract;
+use League\Fractal\Serializer\JsonApiSerializer;
 
 /**
  * Serializer for ProcessMaker API.
  *
  */
-class ApiSerializer extends SerializerAbstract
+class ApiSerializer extends JsonApiSerializer
 {
-    /**
-     * Serialize a collection.
-     *
-     * @param string $resourceKey
-     * @param array $data
-     *
-     * @return array
-     */
-    public function collection($resourceKey, array $data)
-    {
-        if ($resourceKey === false) {
-            return $data;
-        }
-        return array($resourceKey ?: 'data' => $data);
-    }
-
-    /**
-     * Serialize an item.
-     *
-     * @param string $resourceKey
-     * @param array $data
-     *
-     * @return array
-     */
-    public function item($resourceKey, array $data): array
-    {
-        return $data;
-    }
-
-    /**
-     * Serialize null resource.
-     *
-     * @return array
-     */
-    public function null()
-    {
-        return [];
-    }
 
     /**
-     * Serialize the included data.
-     *
-     * @param ResourceInterface $resource
      * @param array $data
      *
-     * @return array
-     *
-     * @codeCoverageIgnore SideloadIncludes not used for this serializer.
+     * @return integer
      */
-    public function includedData(ResourceInterface $resource, array $data): array
+    protected function getIdFromData(array $data)
     {
-        return $data;
+        if (!array_key_exists('uuid', $data)) {
+            throw new InvalidArgumentException(
+            'JSON API resource objects MUST have a valid id'
+            );
+        }
+        return $data['uuid'];
```

Comment:
This line loads the **id** field of the response from **uuid**

Reply:
Where is this method being called from?

Reply:
This used internally by Fractal to get the **id** of the model